src/Core/CommandBus/Parser/CommandHandlerDefinitionParser.php

```diff
@@ -0,0 +1,202 @@
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\CommandBus\Parser;
+
+use ReflectionClass;
+use ReflectionMethod;
+
+class CommandHandlerDefinitionParser
+{
+    private const HANDLER_METHOD_NAME = 'handle';
+    private const RETURN_TAG = '@return';
+
+    /**
+     * @param string $handlerClass
+     * @param string $commandClass
+     *
+     * @return CommandHandlerDefinition
+     */
+    public function parseDefinition(string $handlerClass, string $commandClass): CommandHandlerDefinition
+    {
+        $commandReflection = new ReflectionClass($commandClass);
+        $handlerReflection = new ReflectionClass($handlerClass);
+
+        return new CommandHandlerDefinition(
+            $this->parseType($commandClass),
+            $this->parseDomain($commandClass),
+            $handlerClass,
+            $commandClass,
+            $this->parseCommandConstructorParams($commandReflection),
+            $this->parseDescription($commandReflection),
+            $this->parseReturnType($handlerReflection),
+            $handlerReflection->getInterfaceNames()
+        );
+    }
+
+    /**
+     * @param ReflectionClass $command
+     *
+     * @return string[]
+     */
+    private function parseCommandConstructorParams(ReflectionClass $command): array
+    {
+        if (!$constructor = $command->getConstructor()) {
+            return [];
+        }
+
+        $params = [];
+        foreach ($constructor->getParameters() as $parameter) {
+            $param = sprintf('%s', $parameter->getName());
+            if ($type = $parameter->getType()) {
+                /* @phpstan-ignore-next-line ignoring getName() as method is not visible to phpstan */
+                $param = sprintf('%s %s', $type->getName(), $param);
+            }
+
+            if ($parameter->isOptional()) {
+                $parameter->allowsNull() ? $param = sprintf('?%s', $param) : null;
```

Comment:
The ternary with setter in the middle is a bit weird syntax, not blocking though but something like this would be easier to read:

```suggestion
                $param = $parameter->allowsNull() ? sprintf('?%s', $param) : $param;
```

Reply:
idk, it is equally strange to set $param = $param in else statement for me :sweat_smile: